src/DefinitionResolver.php

```diff
@@ -667,13 +666,21 @@ public function resolveExpressionNodeToType($expr)
                 } else {
                     $classFqn = substr((string)$t->getFqsen(), 1);
                 }
-                $fqn = $classFqn . '->' . $expr->memberName->getText($expr->getFileContents());
+                $add = '->' . $expr->memberName->getText($expr->getFileContents());
                 if ($expr->parent instanceof Node\Expression\CallExpression) {
-                    $fqn .= '()';
+                    $add .= '()';
                 }
-                $def = $this->index->getDefinition($fqn);
-                if ($def !== null) {
-                    return $def->type;
+                $classDef = $this->index->getDefinition($classFqn);
+                if ($classDef !== null) {
+                    foreach ($classDef->getAncestorFQNs($this->index, true) as $fqn) {
```

Comment:
I am not sure I get what you mean - here we need the FQN, so that we can append the property / method part onto it and then look for the definition?